src_c/base.c

```diff
@@ -533,6 +533,79 @@ pg_TwoIntsFromObj(PyObject *obj, int *val1, int *val2)
     return 1;
 }
 
+static int
+pg_IntFromObjEx(PyObject *obj, int *val, const char *msg)
+{
+    if (PyFloat_Check(obj)) {
+        *val = (int)PyFloat_AS_DOUBLE(obj);
+        return 1;
+    }
+
+    *val = PyLong_AsLong(obj);
+    if (PyErr_Occurred()) {
+        if (msg && PyErr_ExceptionMatches(PyExc_TypeError)) {
+            PyErr_SetString(PyExc_TypeError, msg);
+        }
+        return 0;
+    }
+    return 1;
+}
+
+static int
+pg_IntFromSeqIndexEx(PyObject *obj, int index, int *val, const char *msg)
+
+{
+    int result = 0;
+    PyObject *item = PySequence_ITEM(obj, index);
+    if (!item) {
+        return 0;
+    }
+    result = pg_IntFromObjEx(item, val, msg);
+    Py_DECREF(item);
+    return result;
+}
+
+static int
+pg_TwoIntsFromObjEx(PyObject *obj, int *val1, int *val2, const char *msg)
+{
+    /*Faster path for tuples and lists*/
+    if (PyTuple_Check(obj) || PyList_Check(obj)) {
+        Py_ssize_t length = PySequence_Fast_GET_SIZE(obj);
+        PyObject **f_arr = PySequence_Fast_ITEMS(obj);
+        if (length == 2) {
+            if (!pg_IntFromObjEx(f_arr[0], val1, msg) ||
+                !pg_IntFromObjEx(f_arr[1], val2, msg)) {
+                return 0;
+            }
+        }
+        else if (length == 1) {
+            /* Handle case of ((x, y), ) 'sequence inside sequence' */
+            return pg_TwoIntsFromObjEx(f_arr[0], val1, val2, msg);
+        }
+        else {
+            PyErr_SetString(
+                PyExc_ValueError,
+                msg ? msg : "Invalid sequence size, must be either 1 or 2");
+            return 0;
+        }
+    }
+    /*This mainly accounts for vectors, as using PySequence_Fast on a vector is
+     * error-prone*/
+    else if (PySequence_Check(obj)) {
+        if (!pg_IntFromSeqIndexEx(obj, 0, val1, msg))
+            return 0;
+        if (!pg_IntFromSeqIndexEx(obj, 1, val2, msg))
+            return 0;
+    }
```

Comment:
I see some inconsistencies between the tuple/list codepath and the sequence codepath, and ideally behaviour should be same regardless of the type of the sequence.

- The first codepath handles the nested sequence case (which we keep for back-compat) while the second does not. I think both codepaths should be handling this case
- The second codepath does not give clean errors if the size of the sequence is unexpected (but the first does)

Reply:
Thanks! i think i resolved these.